packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -58,20 +58,22 @@ Future<Null> pubGet({
       throwToolExit('pub $command failed ($code)', exitCode: code);
   }
 
-  if (dotPackages.existsSync() && dotPackages.lastModifiedSync().isAfter(pubSpecYaml.lastModifiedSync()))
-    return;
+  if (!dotPackages.existsSync())
+    throwToolExit('$directory: pub did not create .packages file');
 
-  throwToolExit('$directory: pubspec.yaml and .packages are in an inconsistent state');
+  if (dotPackages.lastModifiedSync().isBefore(pubSpecYaml.lastModifiedSync()))
+    throwToolExit('$directory: pub did not update .packages file (pubspec.yaml file has a newer timestamp)');
 }
 
-String _filterOverrideWarnings(String str) {
-  // Warning: You are using these overridden dependencies:
-  // ! analyzer 0.29.0-alpha.0 from path ../../bin/cache/dart-sdk/lib/analyzer
+final RegExp _analyzerWarning = new RegExp(r'^! analyzer [^ ]+ from path \.\./\.\./bin/cache/dart-sdk/lib/analyzer$');
```

Comment:
Is the relative path always the same? For example, if the analyzer override transitive, would it still produce the same path?